File: test/before_train/testset/cpp-java/HumanEval_18_how_many_times.java
import java.util.*;
import java.lang.reflect.*;
import java.security.*;
import java.math.*;
import java.io.*;
import java.util.stream.*;
class Problem {
    public static long howManyTimes(String string, String substring) {
        long times = 0;
        for (int i = 0; i < string.size() - substring.size() + 1; i++) {
            if (string.substr(i, substring.size()) == substring) {
                times += 1;
            }
        }
        return times;
    }
    public static void main(String[] args) {
    assert(howManyTimes((""), ("x")) == (0l));
    assert(howManyTimes(("xyxyxyx"), ("x")) == (4l));
    assert(howManyTimes(("cacacacac"), ("cac")) == (4l));
    assert(howManyTimes(("john doe"), ("john")) == (1l));
    }

}

File: dpo_train/translation_output/java-py/HumanEval_18_how_many_times.py
def how_many_times(string: str, substring: str) -> int:
    return Problem.howManyTimes(string, substring)

def check(candidate):
    assert candidate('', 'x') == 0
    assert candidate('xyxyxyx', 'x') == 4
    assert candidate('cacacacac', 'cac') == 4
    assert candidate('john doe', 'john') == 1

def test_check():
    check(how_many_times)

test_check()
